src/test/java/test/listeners/invokeasinsertionorder/ListenerInvocationAsInsertOrderTest.java

```diff
@@ -0,0 +1,140 @@
+package test.listeners.invokeasinsertionorder;
+
+import static test.listeners.invokeasinsertionorder.Constants.*;
+import java.util.Arrays;
+import java.util.List;
+import org.assertj.core.api.Assertions;
+import org.testng.ITestNGListener;
+import org.testng.TestNG;
+import org.testng.annotations.Test;
+import test.SimpleBaseTest;
+
+public class ListenerInvocationAsInsertOrderTest extends SimpleBaseTest {
```

Comment:
@eyehwan - What happens when you attempt at injecting the listener through the below means ?
1. Via `@Listeners` annotation
2. Via the command line using `-listener`
3. Via a Service Provider Interface as explained in section `5.18.2 - Specifying listeners with ServiceLoader` of https://testng.org/doc/documentation-main.html

Can we add tests for the above 3 mechanisms as well wherein tests try to add multiple listeners via each of the above mechanisms and ensure that the functionality would work fine for listeners that are injected in any manner.

It would also be great if you could add a test that injects listeners using all the 4 mechanisms and establishes an order. That would serve as a good documentation as well to our users since via your delivery we are literally guaranteeing ordering of listeners which wasn't possible earlier and one had to employ something like [this](https://rationaleemotions.com/listener_order/)  (its one of my blogs) to get the ordering sorted out.

Reply:
Have tried with @Listener, which looks fine. 
For the other two ways, I can take a try on it. 

And yes, I can also create a test together with four mechanisms to invoke the listeners.